### PR TITLE
Fix jsonfile cache plugin adds trailing space at end of lines with commas

### PR DESCRIPTION
**Pull request by cuongnv23**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/parsing/utils/jsonify.py

##### ANSIBLE VERSION

```
ansible 2.3.0 (devel 61e6e7868c) last updated 2017/01/21 09:04:05 (GMT +700)
  config file = 
  configured module search path = Default w/o overrides

```

##### SUMMARY
Use `jsonfile` cache plugin with configuration:
```
gathering = smart
fact_caching = jsonfile
fact_caching_connection = /tmp/test
```

The cache file contains trailing space with lines end with comma (`,`), like this:

json_dump

(the `¬` character represents the end of line)

This PR adds `separators` to handle that trailing space. Also simplifies the unit test.

After fixed:
fixed

### CODE REVIEW

**Comment by bcoca** [comment]
what does this break?

File: test/units/parsing/utils/test_jsonify.py [hunk 1]
```diff
@@ -27,9 +27,7 @@ def test_jsonify_simple(self):
         self.assertEqual(jsonify(dict(a=1, b=2, c=3)), '{"a": 1, "b": 2, "c": 3}')
 
     def test_jsonify_simple_format(self):
-        res = jsonify(dict(a=1, b=2, c=3), format=True)
-        cleaned = "".join([x.strip() for x in res.splitlines()])
-        self.assertEqual(cleaned, '{"a": 1,"b": 2,"c": 3}')
+        self.assertEqual(jsonify(dict(a=1, b=2, c=3), format=True), '{\n    "a":1,\n    "b":2,\n    "c":3\n}')
```
Comment (bcoca) [inline test/units/parsing/utils/test_jsonify.py:30]: both are valid tests, i don't see why you need to eliminate the existing one

**Comment by cuongnv23** [comment]
@bcoca current code adds unnecessary spaces, I'm sure that we don't need that

File: test/units/parsing/utils/test_jsonify.py (hunk 1 above)
Comment (cuongnv23) [reply test/units/parsing/utils/test_jsonify.py:30]: yes, both are valid with my PR, as my commit message says, this just simplifies as I don't see the advantage of making the code more complicated.

**Comment by bcoca** [comment]
i still fail to see why this is a problem.

Also why one test is better than the other? I'm fine with adding a new one, but not with removing the old one as they are different tests.

**Comment by cuongnv23** [comment]
I think what we don't need, we should remove. I'm ok to keep the old test, let me remove last commit and update this PR. Thanks

**Comment by cuongnv23** [comment]
my git messed up, close this one, let me send another pr.

**Pull request closed without merging**